demo-terraform/aws/serverless-app/dynamodb.tf

```diff
@@ -2,29 +2,37 @@
 
 ## Dynamodb Table
 resource "aws_dynamodb_table" "basic-dynamodb-table" {
-    name = "${var.dynamodb_table_name}-${random_string.uniq.result}"
-  
-    # Under free tier
-    billing_mode   = "PAY_PER_REQUEST"
-  
-    hash_key       = var.dynamodb_table_pkey
-
-    attribute {
-        name = var.dynamodb_table_pkey
-        type = "S"
-    }
-
-    tags = local.project_tags
+  name = "${var.dynamodb_table_name}-${random_string.uniq.result}"
+
+  # Under free tier
+  billing_mode = "PROVISIONED"
```

Comment:
**Accurics remediation**
 
**[MEDIUM]: Ensure read-write capacities are reserved for AWS DynamoDB tables**
 
**Vulnerability**
Using billing modes other than PROVISIONED may lead to unpredictable surge in cost.
 
**Remediation**
Provisioned mode allows you to specify the number of reads and writes per second that you require for your application. You can use auto scaling to adjust your table’s provisioned capacity automatically in response to traffic changes. This helps you govern your DynamoDB use to stay at or below a defined request rate in order to obtain cost predictability.

source: accurics-